src/test/java/core/basesyntax/service/imp/TransactionParserImpTest.java

```diff
@@ -0,0 +1,85 @@
+package core.basesyntax.service.imp;
+
+import core.basesyntax.model.ProductTransaction;
+import core.basesyntax.service.TransactionParser;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class TransactionParserImpTest {
+    private static final String CORRECT_LINE_BALANCE = "b,apple,20";
+    private static final String CORRECT_LINE_SUPPLY = "s,apple,20";
+    private static final String CORRECT_LINE_PURCHASE = "p,apple,20";
+    private static final String CORRECT_LINE_RETURN = "r,apple,20";
+    private static final String INVALID_TYPE_LINE = "j,apple,20";
+    private static final String ZERO_QUANTITY_LINE = "b,apple,0";
+    private static final String INVALID_RECORDS = "banana, 0";
+    private static final String APPLE = "apple";
+    private static final int QUANTITY = 20;

```

Comment:
It is not really necessary to make a constant for all these values. It would be enough to make a proper name for local vars